### DISCUSSION TITLE
M5 Stack Battery in Home Assistant

### DISCUSSION THREAD

**Discussion by scribblrsam** [HASwitchPlate/openHASP#205] [Q&A]
Hi, 

Apologies, if this is more a question for the HA custom component, but I didn't know if it was more relevant to the discovery code in the main project. 

Is it possible to expose the battery and other stats published by the M5 Stack as sesnsors in Home Assistant? 

I can see the data at `hasp/{name}/state/sensors`. How do I create entities in HA based on this JSON? I know I could delare a sensor with the MQTT integration but that feels like a workaround, equally if its not implemented yet could someone give me some pointers to try and do this and open a PR?

Thanks

**Comment by scribblrsam** [comment]
@nagyrobi @fvanroie @dgomes - Can you point me in the right direction on this?

**Reply by fvanroie** [reply]
> Is it possible to expose the battery and other stats published by the M5 Stack as sesnsors in Home Assistant?

Check out `src\custom\my_custom_template.cpp`. You can add your own key/value pairs to the sensor JSON data:
```
void custom_get_sensors(JsonDocument& doc)
{
    /* Sensor Name */
    JsonObject sensor = doc.createNestedObject(F("Custom"));

    /* Key-Value pair of the sensor value */
    sensor[F("Random")] = random(256);
}
```

What you do with this data is dependent on Home Assistant and custom component.
